pkg/types/types.go

```diff
@@ -58,9 +58,13 @@ func (n *IPNet) UnmarshalJSON(data []byte) error {
 type NetConf struct {
 	Name string `json:"name,omitempty"`
 	Type string `json:"type,omitempty"`
-	IPAM struct {
-		Type string `json:"type,omitempty"`
-	} `json:"ipam,omitempty"`
+	IPAM IPAM   `json:"ipam,omitempty"`
+}
+
+type IPAM struct {
+	Type   string              `json:"type,omitempty"`
+	Subnet string              `json:"subnet"`
```

Comment:
hm actually the only well-known fields for IPAM configuration are Type and Routes (also optional), AIUI, and all others (including subnet) are ipam-plugin-specific. @steveeJ can probably confirm/deny.